src/Nethermind/Nethermind.Synchronization/StateSync/PeerInfoExtensions.cs

```diff
@@ -0,0 +1,28 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using Nethermind.Stats.Model;
+using Nethermind.Synchronization.Peers;
+
+namespace Nethermind.Synchronization.StateSync;
+
+public static class PeerInfoExtensions
+{
+    public static bool CanGetNodeData(this PeerInfo peerInfo)
+    {
+        if (!peerInfo.SyncPeer.Node.AgreedCapability.TryGetValue("eth", out int ethVersion))
```

Comment:
I think you don't need those `AgreedCapabilities` there. 

If we make `ISyncPeer : IProtocolHandler`, which they are, then you can check main protocol and its version without AgreedCapability

Reply:
Or other way to extract main protocol from SyncPeer